docs/audio_driver.md

```diff
@@ -0,0 +1,122 @@
+  |              | Tim6                             | Tim7                   | Tim8  | Tim1_Ch1        |
+  |--------------|----------------------------------|------------------------|-------|-----------------|
+  | dac_basic    | A4+DACD1 = :one:                 | A5+DACD2 = :one:       | state |                 |
+  |              | A4+DACD1 = :one: + Gnd           | A5+DACD2 = :two: + Gnd | state |                 |
+  |              | A4+DACD1 = :two: + Gnd           | A5+DACD2 = :one: + Gnd | state |                 |
+  |              | A4+DACD1 = :one: + Gnd           |                        | state |                 |
+  |              |                                  | A5+DACD2 = :one: + Gnd | state |                 |
+  | dac_additive | A4+DACD1 = :one: + Gnd           |                        |       |                 |
+  |              | A5+DACD2 = :one: + Gnd           |                        |       |                 |
+  |              | A4+DACD1 + A5+DACD2 = :one: [^2] |                        |       |                 |
+  | pwm_software | state-update                     |                        |       | any = :one:     |
+  | pwm hardware | state-update                     |                        |       | A8 = :one: [^3] |
+
+
+[^1]: the routing and alternate functions for PWM differ sometimes between STM32 MCUs, if in doubt consult the data-sheet
+[^2]: one piezo connected to A4 and A5, with AUDIO_PIN_ALT_AS_NEGATIVE set
+[^3]: TIM1_CH1 = A8 on STM32F103C8, other combinations are possible, see Data-sheet. configured with: AUDIO_PWM_TIMER and AUDIO_PWM_TIMERCHANNEL
+
+
+
+### DAC basic
+
+The default driver for ARM boards, in absence of an overriding configuration.
+This driver needs one Timer per enabled/used DAC channel, to trigger conversion; and a third timer to trigger state updates with the audio-core.
+
+
+### DAC additive
+
+only needs one timer (GPTD6, Tim6) to trigger the DAC unit to do a conversion; the audio state updates are in turn triggered during the DAC callback.
+
+
+### PWM hardware
+
+this driver uses the ChibiOS-PWM system to produce a square-wave on specific output pins that are connected to the PWM hardware.
+The hardware directly toggles the pin via its alternate function. see your MCUs data-sheet for which pin can be driven by what timer - looking for TIMx_CHy and the corresponding alternate function.
+
+a configuration example for the STM32F103C8 would be:
+``` c
+//halconf.h:
+#define HAL_USE_PWM                 TRUE
+#define HAL_USE_PAL                 TRUE
+#define HAL_USE_GPT                 TRUE
+```
+
+``` c
+// mcuconf.h:
+#define STM32_PWM_USE_TIM1                  TRUE
+#define STM32_GPT_USE_TIM3                  TRUE
+```
+
+If we now target pin A8, looking trough the data-sheet of the STM32F103C8, for the timers and alternate functions
+TIM1_CH1 = PA8 <- alternate0
+TIM1_CH2 = PA9
+TIM1_CH3 = PA10
+TIM1_CH4 = PA11
```

Comment:
Are you intending for this to be a table or list?

Reply:
converted to unordered list - thanks!